table/block_based/block_based_table_reader.cc

```diff
@@ -1376,7 +1378,7 @@ Status BlockBasedTable::ReadPropertiesBlock(
         rep_->index_type == BlockBasedTableOptions::kBinarySearchWithFirstKey;
 
     s = GetGlobalSequenceNumber(*(rep_->table_properties), largest_seqno,
-                                &(rep_->global_seqno));
+                                has_global_seqno, &(rep_->global_seqno));
```

Comment:
I didn't see anything changed in binary search dealing with seqno. Right now, binary search should still work because all keys is encoded with seqno, so when we seek to (user_key, seqno), we are guaranteed to see all entries with user_key. It's not the case if seqno is not always 0. Something needs to be done here to make it work.

Reply:
I think all seek case would be seek(user_key, MAX_SEQ). If not I could change it.

Reply:
OK, I would change it.